pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -200,31 +202,69 @@ public String recommendConfig(String inputStr) {
   @Produces(MediaType.APPLICATION_JSON)
   @Path("/tables")
   @ApiOperation(value = "Lists all tables in cluster", notes = "Lists all tables in cluster")
-  public String listTableConfigs(@ApiParam(value = "realtime|offline") @QueryParam("type") String tableTypeStr) {
+  public String listTables(@ApiParam(value = "realtime|offline") @QueryParam("type") String tableTypeStr,
+      @ApiParam(value = "name|creationTime|lastModifiedTime") @QueryParam("sortType") @DefaultValue("name") String sortType,
+      @ApiParam(value = "true|false") @QueryParam("sortAsc") @DefaultValue("true") boolean sortAsc) {
     try {
       List<String> tableNames;
       TableType tableType = null;
       if (tableTypeStr != null) {
         tableType = TableType.valueOf(tableTypeStr.toUpperCase());
       }
 
+      Boolean sortTypeValid = sortType.equalsIgnoreCase("name") || sortType.equalsIgnoreCase("creationTime") || sortType
+          .equalsIgnoreCase("lastModifiedTime");
+      if (!sortTypeValid) {
+        throw new IllegalArgumentException(
+            "sortType expects={name|creationTime|lastModifiedTime} " + "got " + sortType);
+      }
+
       if (tableType == null) {
         tableNames = _pinotHelixResourceManager.getAllRawTables();
+      } else if (tableType == TableType.REALTIME) {
+        tableNames = _pinotHelixResourceManager.getAllRealtimeTables();
       } else {
-        if (tableType == TableType.REALTIME) {
-          tableNames = _pinotHelixResourceManager.getAllRealtimeTables();
-        } else {
-          tableNames = _pinotHelixResourceManager.getAllOfflineTables();
-        }
+        tableNames = _pinotHelixResourceManager.getAllOfflineTables();
       }
 
-      Collections.sort(tableNames);
+      HashMap<String, TableStats> MapStats = new HashMap<String, TableStats>();
+      for (String tableName : tableNames) {
+        MapStats.put(tableName, _pinotHelixResourceManager.getTableStats(tableName));
+      }
+
```

Comment:
Based on the sort type, sort accordingly:
```suggestion
  if (sortType == null || sortType.equalsIgnoreCase("name")) {
    tableNames.sort(sortAsc ? null : Comparator.reverseOrder());
  } else {
    boolean sortByCreationTime = sortType.equalsIgnoreCase("creationTime");
    Preconditions.checkState(sortByCreationTime || sortType.equalsIgnoreCase("lastModifiedTime"), ...);
    
    List<TableStats> tableStatsList = new ArrayList<>(tableNames.size());
    for (String tableName : tableNames) {
      tableStatsList.add(_pinotHelixResourceManager.getTableStats(tableName));
    }
    
    int sortFactor = sortAsc ? 1 : -1;
    if (sortByCreationTime) {
      Collections.sort(tableStatsList, (o1, o2) -> o1.getCreationTime().compareTo(o2.getCreationTime()) * sortFactor;
    } else {
      Collections.sort(tableStatsList, (o1, o2) -> o1.getModifiedTime().compareTo(o2.getModifiedTime()) * sortFactor;
    }
    
    // Create a new list from the sorted stats...
  }
```

Reply:
I have refactored the code, please let me know if this makes sense